internal/pkg/upgrade/upgrade.go

```diff
@@ -46,6 +62,44 @@ func NewUpgrade(url string) (err error) {
 	return err
 }
 
+func leaveEtcd() (err error) {
+	tlsInfo := transport.TLSInfo{
+		CertFile:      "/etc/kubernetes/pki/etcd/peer.crt",
```

Comment:
Curious if you can make use of any of the constants we use here https://github.com/talos-systems/talos/blob/master/internal/pkg/constants/constants.go#L96

Reply:
Do we have access to these certs inside osd or would we need to shuffle this to init?

Reply:
Upgrades are proxied through to the `init` API so we should have access to all the keys we need.